ui/src/components/Map/MapNodesGrid.vue

```diff
@@ -0,0 +1,215 @@
+<template>
+  <div class="mapnodes">
+    <div class="button-group">
+      <span class="buttons">
+        <feather-button primary v-on:click="clearFilters()">Clear Filters</feather-button>
+        <feather-button primary v-on:click="confirmFilters()">Filter Map</feather-button>
+        <feather-button primary v-on:click="reset()">Reset</feather-button>
+      </span>
+    </div>
+    <div class="map-nodes-grid">
+      <ag-grid-vue
+        style="width: 100%; height: 700px"
+        class="ag-theme-alpine"
+        rowSelection="multiple"
+        :columnDefs="columnDefs"
+        @grid-ready="onGridReady"
+        :rowData="rowData"
+        :defaultColDef="defaultColDef"
+        :gridOptions="gridOptions"
+        :pagination="true"
+        @rowDoubleClicked="rowDoubleClicked"
+      ></ag-grid-vue>
+    </div>
+  </div>
+</template>
+<script setup lang="ts">
+import { ref } from "vue";
+import "ag-grid-community/dist/styles/ag-grid.css";
+import "ag-grid-community/dist/styles/ag-theme-alpine.css";
+import { AgGridVue } from "ag-grid-vue3";
+import { useStore } from "vuex";
+import { computed, watch } from 'vue'
+import { Coordinates, Node } from "@/types";
+import { FeatherButton } from "@featherds/button";
+
+const store = useStore();
+
+const gridOptions = ref({})
+
+const defaultColDef = ref({
+  floatingFilter: true,
+  resizable: true,
+  enableBrowserTooltips: true,
+  filter: "agTextColumnFilter",
+  sortable: true,
+  suppressMenu: true
+})
+
+let interestedNodesID = computed(() => {
+  return store.getters['mapModule/getInterestedNodesID'];
+})
+
+let rowData = ref(getGridRowDataFromInterestedNodes());
+
+function getGridRowDataFromInterestedNodes() {
+  return store.getters['mapModule/getInterestedNodes'].map((node: any) => ({
+    id: +node.id,
+    foreignSource: node.foreignSource,
+    foreignId: node.foreignId,
+    label: node.label,
+    labelSource: node.labelSource,
+    lastCapabilitiesScan: node.lastCapsdPoll,
+    primaryInterface: node.primaryInterface,
+    sysObjectid: node.sysObjectId,
+    sysName: node.sysName,
+    sysDescription: node.sysDescription,
+    sysContact: node.sysContact,
+    sysLocation: node.sysLocation
+  }));
+}
+
+let gridApi: any = ref({});
+let gridColumnApi: any = ref({});
+
+function onGridReady(params: any) {
+  gridApi = params.api
+  gridColumnApi = params.columnApi
+  autoSizeAll(false);
+}
+
+function autoSizeAll(skipHeader: boolean) {
+  var allColumnIds: string[] = [];
+  gridColumnApi.getAllColumns().forEach(function (column: any) {
+    allColumnIds.push(column.colId);
+  });
+  gridColumnApi.autoSizeColumns(allColumnIds, skipHeader);
+}
+
+watch(
+  () => interestedNodesID.value,
+  () => {
+    gridApi.setRowData(
+      getGridRowDataFromInterestedNodes()
+    );
```

Comment:
Getting an error here when page first loads `gridApi.setRowData is not a function`. On refreshing it works though, so looks like it's just not ready on the initial load. Might only need to wrap it in a conditional to see if it's available.

Reply:
Thanks for finding this. I added a condition to check if ``gridApi.setRowData `` is available.